### PR TITLE
removing setitem since it's not required, deep copy via cast, unique,…

### PR DESCRIPTION
**Pull request by artemru**
… combine_chunks, ... test will come soon

### CODE REVIEW

File: fletcher/base.py [hunk 1]
```diff
@@ -191,53 +191,39 @@ def __init__(self, array, dtype=None, copy=None):
...
         Returns
         -------
-        size : int
+        length : int
         """
         # type: () -> int
         return len(self.data)
 
-    @property
-    def shape(self):
```
Comment (artemru) [inline fletcher/base.py:221]: shape and ndim are already present on parent class.

File: fletcher/base.py [hunk 2]
```diff
@@ -313,90 +280,6 @@ def _reduce(self, name, skipna=True, **kwargs):
             )
         )
 
-    def __setitem__(self, key, value):
```
Comment (artemru) [inline fletcher/base.py:316]: this's not required by pandas ! it looks strange to want to modify elements of immutable data structure.

File: fletcher/base.py [hunk 3]
```diff
@@ -487,22 +370,17 @@ def copy(self, deep=False):
         -------
         ExtensionArray
         """
-        if deep:
-            raise NotImplementedError("Deep copy is not supported")
-        return type(self)(self.data)
+        data = self.data.cast(self.data.type) if deep else self.data
```
Comment (artemru) [inline fletcher/base.py:378]: `cast` will make a copy !

File: fletcher/base.py [hunk 4]
```diff
@@ -487,22 +370,17 @@ def copy(self, deep=False):
...
         """
         The number of bytes needed to store this object in memory.
         """
-        size = 0
-        for chunk in self.data.chunks:
-            for buf in chunk.buffers():
-                if buf is not None:
-                    size += buf.size
-        return size
+        return sum(buf.size for chunk in self.data.chunks
```
Comment (artemru) [inline fletcher/base.py]: just a refactor !

File: fletcher/base.py [hunk 5]
```diff
@@ -511,6 +389,9 @@ def base(self):
         """
         return self.data
 
+    def unique(self):
```
Comment (artemru) [inline fletcher/base.py:401]: as simple as that !!

**Comment by artemru** [comment]
@krivonogov, any idea why tests are not starting ? is there some linter on this ?

File: fletcher/base.py [hunk 6]
```diff
@@ -191,53 +191,39 @@ def __init__(self, array, dtype=None, copy=None):
                     self.__class__.__name__, type(array)
                 )
             )
+        self.data.validate()
```
Comment (krivonogov) [inline fletcher/base.py]: there was linter error (`black --check`: https://github.com/psf/black) before, I have fixed it (it is enough to launch `black name_of_file.py` locally), now it breaks here, as this method is not available on `ChunkedArray`

Comment (artemru) [reply fletcher/base.py]: thanks !!

Comment (artemru) [reply fletcher/base.py]: ok, `chunked_array.validate()` only available starting from 0.15.

**Comment by krivonogov** [comment]
maybe we can split this PR between
* simple cleaning
* removing `setitem` (I guess this one is blocked by refactor of `fillna`)
* refactor of `factorize`

@artemru wdyt?

**Pull request closed without merging**